board/rt1051/crashdump/crashdumpwriter_vfs.cpp

```diff
@@ -41,6 +56,21 @@ namespace crashdump
         std::fclose(file);
     }
 
+    void CrashDumpWriterVFS::deleteOldDumps()
+    {
+        std::vector<std::filesystem::path> crashdumps{};
+        for (const auto &entry : std::filesystem::directory_iterator(purefs::dir::getCrashDumpsPath())) {
+            std::cout << entry.path() << std::endl;
+            crashdumps.push_back(entry.path());
+        }
+
+        while (not(crashdumps.size() <= maxFilesCount)) {
+            LOG_INFO("Deleting %s ...", crashdumps.front().c_str());
+            std::remove(crashdumps.front().c_str());
```

Comment:
What is this line exactly doing? Is it removing all the items equal to the first one?

Also: is the purpose of this loop to remove some items so that the new size doesn't exceed `maxFilesCount`? Is yes, then I think doing a little computation and removing the whole "tail" basing on iterators would be much better.